aws/internal/keyvaluetags/key_value_tags.go

```diff
@@ -0,0 +1,171 @@
+//go:generate go run generators/listtags/main.go
+//go:generate go run generators/servicetags/main.go
+//go:generate go run generators/updatetags/main.go
+
+package keyvaluetags
+
+import (
+	"strings"
+)
+
+const AwsTagKeyPrefix = `aws:`
+const ElasticbeanstalkTagKeyPrefix = `elasticbeanstalk:`
+const NameTagKey = `Name`
```

Comment:
Also, [RDS](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_Tagging.html#Overview.Tagging) and [Neptune](https://docs.aws.amazon.com/neptune/latest/userguide/tagging.html#Overview.Tagging) tag keys cannot start with `rds:`.

Reply:
Excellent find! The current functions for these two services are not currently handling this:

- https://github.com/terraform-providers/terraform-provider-aws/blob/b4efb0be0ac4315f2f4ac52955e58fc32ffe05d5/aws/tagsRDS.go#L123
- https://github.com/terraform-providers/terraform-provider-aws/blob/b4efb0be0ac4315f2f4ac52955e58fc32ffe05d5/aws/tagsNeptune.go#L106

Given that there are now a few services that require some special tag key handling, maybe it'll be best to these service-specific additions directly into the `IgnoreAws()` function directly instead of creating bespoke custom functions, which may be missed during implementations.

Reply:
Ah, in order encode each service's customizations into a single `IgnoreAws()` function, the `KeyValueTags` type would need to know the service name of the tags it represents. It is certainly doable, but it would increase the implementation complexity considerably for little benefit at this time. This implementation now includes a separate `IgnoreRds()` function for the `KeyValueTags` type which of course can always be refactored later, should this type of implementation be really problematic in the future. 👍